openff/recharge/cli/reconstruct.py

```diff
@@ -39,30 +34,17 @@ def _retrieve_result_records(
 
     results = client.query_records(
         record_id=record_ids,
-        limit=client.server_info["api_limits"]["get_records"],
     )
 
-    # Fetch the corresponding record keywords.
-    # The iterator is depleted on its first unpacking, so need to re-query
-    keywords = [
-        result.specification.keywords
-        for result in client.query_records(
-            record_id=record_ids,
-            limit=client.server_info["api_limits"]["get_records"],
-        )
-    ]
-
-    return results, keywords
+    return results
 
 
 def _process_result(
-    result_tuple: tuple[
-        "qcportal.record_models.BaseRecord",
-        "qcelemental.models.Molecule",
-        "qcportal.models.KeywordSet",
-    ],
+    qc_result: "qcportal.record_models.BaseRecord",
     grid_settings: GridSettingsType,
 ):
+    result_tuple = (qc_result, qc_result.molecule, qc_result.specification.keywords)
```

Comment:
I think we can avoid this intermediate tuple and pass the values straight to the `from_qcportal_results` function.